base_layer/wallet/src/transaction_service/protocols/transaction_validation_protocol.rs

```diff
@@ -214,23 +214,13 @@ where
         );
         let op_id = self.operation_id;
         while let Some(last_mined_transaction) = self.db.fetch_last_mined_transaction().for_protocol(op_id)? {
+            // we can use expect here as fetch_last_mined_transaction filters on mined_height and mined_in block is some
             let mined_height = last_mined_transaction
                 .mined_height
-                .ok_or_else(|| {
-                    TransactionServiceError::ServiceError(
-                        "fetch_last_mined_transaction() should return a transaction with a mined_height".to_string(),
-                    )
-                })
-                .for_protocol(op_id)?;
+                .expect("Tx validation mined height is missing");
```

Comment:
I do not believe this is an improvement. In my mind the previous code that returned an error was better, but still not ideal.

I would prefer that we just log an error message and continue to the next _last mined transaction_ should we have data inconsistencies. This way the protocol will not stop and do an endless repeat should we have a single row with problematic data.

Reply:
I removed this as the TMS defines a mined transaction as a tx with a `mined_height` and `mined_block`. 
The above query we ask to get us a transaction with these properties explicitly asks the db for this. 

See: https://github.com/tari-project/tari/blob/92ceef998dc25392b6c76e2d4a86120eff5ebc33/base_layer/wallet/src/transaction_service/storage/sqlite_db.rs#L1069

So the only reason this can be None is if somehow the memory got corrupted between asking the db and us now getting the value.

Reply:
this can't be an expect